### ISSUE TITLE
`extra_failure_lines` are wrapped with blank lines multiple times for aggregated failures

### ISSUE THREAD

**Issue by ilyazub** [rspec/rspec-core#3005]
### Subject of the issue

[`RSpec::Core::Formatters::ExceptionPresenter#extra_failure_lines`](https://github.com/rspec/rspec-core/blob/a27c169f7ea1b3851c6b7ea8d9c910ccdcb6725c/lib/rspec/core/formatters/exception_presenter.rb#L196-L205) wraps `lines` with unexpected blank lines. Number of wrapped blank lines equals to an index of an aggregated failure.

### Your environment
* Ruby version: ruby 3.0.2p107 (2021-07-07) [x86_64-linux]
* rspec-core version: 3.12.0

### Steps to reproduce

Code in online IDE: https://replit.com/@ilyazub/rspec-extra-failure-lines-blank-lines#rspec_report.rb

```ruby
# frozen_string_literal: true

begin
  require 'bundler/inline'
rescue LoadError => e
  warn 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'

  gem 'rspec', '3.12.0'
end

puts "Ruby version is: #{RUBY_VERSION}"
require 'rspec/autorun'

class ExampleListener
  def example_failed(notification)
    example = notification.example
    exception = example.exception

    return unless exception

    lines = example.metadata[:extra_failure_lines] = Array(example.metadata[:extra_failure_lines])

    lines << 'booom'

    # lines are appended only once
    # Uncomment and execute to double-check
    # puts "extra_failure_lines appended"
  end
end

RSpec.configure do |c|
  c.reporter.register_listener ExampleListener.new, :example_failed
end

RSpec.describe 'blank lines on aggregated failures with extra_failure_lines', :aggregate_failures do
  it 'fails' do
    (1..10).each do |_i|
      RSpec::Expectations.fail_with 'tadaa'
    end
  end
end
```

### Expected behavior

`extra_failure_lines` are wrapped once per example.

```
     1.3) Failure/Error: RSpec::Expectations.fail_with 'tadaa'
            tadaa
          
          booom
          
          # rspec_report.rb:39:in `block (3 levels) in <main>'
          # rspec_report.rb:38:in `each'
          # rspec_report.rb:38:in `block (2 levels) in <main>'

     1.10) Failure/Error: RSpec::Expectations.fail_with 'tadaa'
             tadaa
           
           booom
           
           # rspec_report.rb:39:in `block (3 levels) in <main>'
           # rspec_report.rb:38:in `each'
           # rspec_report.rb:38:in `block (2 levels) in <main>'
```

### Actual behavior

`extra_failure_lines` are wrapped number of times equals to the number of aggregated failures. Number of wrapped blank lines equals to an index of an aggregated failure.

```

     1.3) Failure/Error: RSpec::Expectations.fail_with 'tadaa'
            tadaa
          
          
          
          booom
          
          
          
          # rspec_report.rb:39:in `block (3 levels) in <main>'
          # rspec_report.rb:38:in `each'
          # rspec_report.rb:38:in `block (2 levels) in <main>'

     1.10) Failure/Error: RSpec::Expectations.fail_with 'tadaa'
             tadaa
           
           
           
           
           
           
           
           
           
           
           booom
           
           
           
           
           
           
           
           
           
           
           # rspec_report.rb:39:in `block (3 levels) in <main>'
           # rspec_report.rb:38:in `each'
           # rspec_report.rb:38:in `block (2 levels) in <main>'
```

**Comment by JonRowe** [comment]
Fix released in `3.12.1`

**Comment by JonRowe** [comment]
It was released on the 3rd of Feburary.
